packages/flame_network_assets/lib/src/flame_network_assets.dart

```diff
@@ -0,0 +1,195 @@
+/// cache layers before making the http request, if both layer are cache miss,
+/// then the request is made and both layers set with the response.
+///
+/// Another important note about the cache layers is that the first layer, is a
+/// per instance cache, while the local storage is an app global cache. This
+/// means that two different [FlameAssetResponse] instances will have the same
+/// local storage cache, but not the same memory cache.
+///
+/// Note that the local storage layer is not present when running on web since
+/// that platform doesn't really have a file system. The browser caching will
+/// work as a similar replacement for this layer, though that can't be
+/// controlled by this package, make sure that the server where the images
+/// are being fetched returns the correct cache header to make the browser
+/// cache the assets.
+///
+/// Each cache layer can be disabled by the [cacheInMemory] or [cacheInStorage]
+/// argument on the constructor.
+///
+/// {@endtemplate}
+abstract class FlameNetworkAssets<T> {
+  /// {@macro flame_network_assets}
+  ///
+  /// - [decodeAsset] a [DecodeAssetFunction] responsible for decoding the asset
+  /// from its raw format.
+  /// - [encodeAsset] a [EncodeAssetFunction] responsible for encoding the asset
+  /// to its raw format.
+  /// - [get] is an optional [GetAssetFunction], if omitted [http.get] is used
+  /// by default.
+  /// - [getAppDirectory] is an optional [GetAppDirectoryFunction], if omitted
+  /// [getApplicationDocumentsDirectory] is used by default.
+  /// - [cacheInMemory] will not cache assets in the memory when false,
+  /// (true by default).
+  /// - [cacheInStorage] will not cache assets in the file system when false,
+  /// (true by default).
+  FlameNetworkAssets({
+    required DecodeAssetFunction<T> decodeAsset,
+    required EncodeAssetFunction<T> encodeAsset,
+    GetAssetFunction? get,
+    GetAppDirectoryFunction? getAppDirectory,
+    this.cacheInMemory = true,
+    this.cacheInStorage = true,
+  })  : _isWeb = kIsWeb,
+        _decode = decodeAsset,
+        _encode = encodeAsset {
+    _get = get ??
+        (
+          String url, {
+          Map<String, String>? headers,
+        }) =>
+            http.get(Uri.parse(url), headers: headers).then((response) {
+              return FlameAssetResponse(
+                statusCode: response.statusCode,
+                bytes: response.bodyBytes,
+              );
+            });
+
+    _getAppDirectory = getAppDirectory ?? getApplicationDocumentsDirectory;
+  }
+
+  late final GetAssetFunction _get;
+  late final GetAppDirectoryFunction _getAppDirectory;
+  final DecodeAssetFunction<T> _decode;
+  final EncodeAssetFunction<T> _encode;
+
+  /// Flag indicating if files will be cached in memory.
+  final bool cacheInMemory;
+
+  /// Flag indicating if files will be cached in the local storage.
+  final bool cacheInStorage;
+
+  final bool _isWeb;
```

Comment:
can we just set the default value of cacheInStorage to be kIsWeb instead of having two flags?